rs/index/src/collection/collection.rs

```diff
@@ -67,6 +69,11 @@ pub struct Collection<Q: Quantizer + Clone + Send + Sync> {
 
 impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
     pub fn new(base_directory: String, segment_config: CollectionConfig) -> Result<Self> {
+        let collection_name = base_directory
+            .split('/')
+            .last()
+            .unwrap_or("unknown")
+            .to_string();
```

Comment:
let's not do this. We should be able to just point to a random base_directory. Let's pass the `collection_name` to the `new` function